cmd/api/routers/routers.go

```diff
@@ -30,7 +43,14 @@ func (c *Controller) GetAllResources(context *gin.Context) {
 		context.JSON(http.StatusInternalServerError, err.Error())
 	}
 
-	context.JSON(http.StatusOK, resources)
+	apiVersion := fmt.Sprintf("%s/%s", group, version)
+	listKind := fmt.Sprintf("%sList", kind)
```

Comment:
I have some concerns about  building this variable with just appending the suffix "List" to the kind...

When a CRD is created, you could optionally provide the field `spec.names.listKind`. By default this value will be "[kind]List" but it could potentially be a different one (I have never seen a different one, but the possibility is there). With this command the `kindList` parameter for each CRD can be listed: `kubectl get crd -o jsonpath='{.items[*].spec.names.listKind}'`

At first I thought that this parameter should be provided by the [APIResource struct](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1141) that we have stored in the context by the GetAPIResources middleware, but AFAICS, it isn't there...

I guess that we could query the CRD and retrieve the `kindList` field but I don't like the idea of a new query for requesting that information... so I prefer to stick with this code, but maybe we could add a comment saying that this could potentially not match the listKind of a CRD

WDYT?

Reply:
Now that you mention it... Then I would return a generic `List` and see what happens, if someone complains, then we need to find a way of retrieving the information. If no one complains, then we don't need to think twice. I don't want to have another query to retrieve this either.

Reply:
Good idea, I think that approach would be better

Reply:
This sounds dangerous to me.  Why not just get the real list kind and never have to worry about it again?